src/ssa_tracker.rs

```diff
@@ -88,11 +87,10 @@ impl SSATracker {
     pub fn introduce_variable_of_type(
         &mut self,
         variable: &Identifier,
-        type_: String,
+        type_: SMTSort,
     ) -> SMTVariable {
-        let var = self.introduce_variable(variable);
         self.types.insert(self.identifier_to_id(variable), type_);
-        var
+        self.introduce_variable(variable)
```

Comment:
This was actually a bug. The type being inserted after meant that the returned `var` actually didn't have the type info. Fixed here.